### PR TITLE
ci: test examples

### PR DESCRIPTION
**Pull request by cpcloud**
We were not testing examples in CI because we did not install the `examples` extra. Now we are.

### CODE REVIEW

File: ibis/backends/datafusion/__init__.py [hunk 1]
```diff
@@ -372,7 +373,7 @@ def _register(
             self.con.deregister_table(table_name)
             self.con.register_record_batches(table_name, [[source]])
             return self.table(table_name)
-        elif isinstance(source, pa.dataset.Dataset):
+        elif isinstance(source, ds.Dataset):
```
Comment (ncclementi) [inline ibis/backends/datafusion/__init__.py:376]: This seems to be unrelated to including examples in the tests suit, just making sure you are ok with that.

File: ibis/backends/datafusion/tests/test_register.py [hunk 2]
```diff
@@ -45,6 +44,8 @@ def test_register_batches(conn):
 
 
 def test_register_dataset(conn):
+    import pyarrow.dataset as ds
```
Comment (ncclementi) [inline ibis/backends/datafusion/tests/test_register.py:47]: same that the comment above. If we are good with these nits getting in here, all good.

**Review by ncclementi — APPROVED** [review]
LGTM, just two small comments to check weather those changes belong here or not.

File: ibis/backends/datafusion/__init__.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/datafusion/__init__.py:376]: These are definitely related. They aren't hit unless the examples are run.

File: ibis/backends/datafusion/tests/test_register.py (hunk 2 above)
Comment (cpcloud) [reply ibis/backends/datafusion/tests/test_register.py:47]: Definitely related, for the same reason.
